app/controllers/concerns/api_rescuable.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module ApiRescuable
+  extend ActiveSupport::Concern
+
+  included do
```

Comment:
Add `rescue_from Exception, with: :handle_exception` to handle general errors. Please also check that this doesn't take priority over other handler(defining this as first filter should do).